collector/collector_test.go

```diff
@@ -300,6 +309,158 @@ info:
 	})
 
 	Describe("Scan", func() {
+		Context("When users are created for the same stage on different files (issue kairos-io/kairos#1341)", func() {
+			var cmdLinePath, tmpDir1 string
+			var err error
+
+			BeforeEach(func() {
+				tmpDir1, err = os.MkdirTemp("", "config1")
+				Expect(err).ToNot(HaveOccurred())
+				err := os.WriteFile(path.Join(tmpDir1, "local_config_1.yaml"), []byte(`#cloud-config
+install:
+  auto: true
+  reboot: false
+  poweroff: false
+  grub_options:
+     extra_cmdline: "console=tty0"
+options:
+  device: /dev/sda
+stages:
+  initramfs:
+    - users:
+        kairos:
+          groups:
+            - sudo
+          passwd: kairos
+`), os.ModePerm)
+				Expect(err).ToNot(HaveOccurred())
+				err = os.WriteFile(path.Join(tmpDir1, "local_config_2.yaml"), []byte(`#cloud-config
+stages:
+  initramfs:
+    - users:
+        foo:
+          groups:
+            - sudo
+          passwd: bar
+`), os.ModePerm)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				err = os.RemoveAll(tmpDir1)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("keeps the two users", func() {
+				o := &Options{}
+				err := o.Apply(
+					MergeBootLine,
+					WithBootCMDLineFile(cmdLinePath),
+					Directories(tmpDir1),
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				c, err := Scan(o, FilterKeysTestMerge)
+				Expect(err).ToNot(HaveOccurred())
+
+				fmt.Println(c.String())
+				Expect(c.String()).To(Equal(`#cloud-config
+
+install:
+    auto: true
+    grub_options:
+        extra_cmdline: console=tty0
+    poweroff: false
+    reboot: false
+options:
+    device: /dev/sda
+stages:
+    initramfs:
+        - users:
+            kairos:
+                groups:
+                    - sudo
+                passwd: kairos
+        - users:
+            foo:
+                groups:
+                    - sudo
+                passwd: bar
+`))
+			})
+		})
+
+		Context("When a YIP if expression is contained", func() {
```

Comment:
add test for the reporting issue https://github.com/kairos-io/kairos/issues/1665